service/controller/resource/monitoring/prometheus/resource.go

```diff
@@ -172,11 +172,8 @@ func toPrometheus(ctx context.Context, v interface{}, config Config) (metav1.Obj
 								{
 									MatchExpressions: []corev1.NodeSelectorRequirement{
 										{
-											Key:      "role",
-											Operator: corev1.NodeSelectorOpNotIn,
-											Values: []string{
-												"control-plane",
-											},
+											Key:      "node-role.kubernetes.io/control-plane",
+											Operator: corev1.NodeSelectorOpDoesNotExist,
```

Comment:
Prometheus was scheduled onto a master node because role is set to master and not control-plane:

kubernetes.io/role: master
node-role.kubernetes.io/control-plane: ""
node-role.kubernetes.io/master: ""
role: master